app/models/image.rb

```diff
@@ -35,4 +35,33 @@ def crop_variant(resize = "#{WIDTH}x#{HEIGHT}")
       resize: resize,
     )
   end
+
+  def cropped_file
+    image = thumbnail.processed
+    io = StringIO.new(image.service.download(image.key))
+    AssetManagerFile.new(io, filename, blob.content_type, asset_manager_file_url)
+  end
+
+  # Used as a stand-in for a File / Rack::Multipart::UploadedFile object when
+  # passed to GdsApi::AssetManager#create_asset. The interface is required for
+  # uploading a file using the restclient we used in the backend.
+  #
+  # https://github.com/rest-client/rest-client/blob/master/lib/restclient/payload.rb#L181-L194
+  #
+  # This is done by delegating to 'io' which should implement the IO interface
+  # (e.g. StringIO) and adds methods to return filename, content_type and path.
+  class AssetManagerFile < SimpleDelegator
+    attr_reader :filename, :content_type, :url
```

Comment:
I think you want filename to be original_filename as per: https://www.rubydoc.info/gems/rack/Rack/Multipart/UploadedFile